subsys/bluetooth/host/scan.c

```diff
@@ -209,7 +209,7 @@ static int start_le_scan_legacy(uint8_t scan_type, uint16_t interval, uint16_t w
 	set_param.interval = sys_cpu_to_le16(interval);
 	set_param.window = sys_cpu_to_le16(window);
 
-	if (IS_ENABLED(CONFIG_BT_WHITELIST) &&
+	if (IS_ENABLED(CONFIG_BT_FA_LIST) &&
 	    atomic_test_bit(bt_dev.flags, BT_DEV_SCAN_WL)) {
```

Comment:
```suggestion
	    atomic_test_bit(bt_dev.flags, BT_DEV_SCAN_FL)) {
```

Reply:
Agree

Reply:
This PR only changes the public API @LingaoM. We will need further cleanups, but I want to keep those separate.